File: frontend/src/app/_model/_interface/quality-checks.ts
import { Cars } from './car';
import { Employee } from './employee';

export interface QualityChecks {
  id: string;
  check_date: string;
  passed: boolean;
  notes: string;
  car_id: Cars;
  inspector_id: Employee;
}

export function isQualityCheck(obj: any): obj is QualityChecks {
  return (
    obj &&
    typeof obj.id === 'string' &&
    typeof obj.check_date === 'string' &&
    typeof obj.passed === 'boolean' &&
    typeof obj.notes === 'string' &&
    obj.car_id !== undefined &&
    obj.inspector_id !== undefined
  );
}

File: backend/src/main/java/com/example/backend/Model/Class/QualityChecks.java
package com.example.backend.Model.Class;

import com.example.backend.Configuration.BlobSerializer;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import lombok.*;

import java.sql.Blob;
import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Getter
@Setter
@Entity
@Cacheable
@Table(name = "QUALITY_CHECKS")
public class QualityChecks {
    @Id
    private String id;
    @Column(name = "check_date")
    private LocalDateTime check_date;
    @Column(name = "passed")
    private Boolean passed;
    @Lob
    @Column(name = "notes")
    @JsonSerialize(using = BlobSerializer.class)
    private Blob notes;
    @ManyToOne
    @JoinColumn(name = "car_id", referencedColumnName = "id")
    private Cars car_id;
    @ManyToOne
    @JoinColumn(name = "inspector_id", referencedColumnName = "id")
    private Employees inspector_id;
}
